robot:

  joint_trajectory_controller:
    type: effort_controllers/JointTrajectoryController
    publish_rate: 20
    joints:
      - joint1_f
      - joint2_f
      - joint1_t
      - joint2_t

    gains: # Required because we're controlling an effort interface
      joint1_f: {p: 32,  d: 0, i: 0.1, i_clamp: 1}
      joint2_f: {p: 32,  d: 0, i: 0.1, i_clamp: 1}
      joint1_t: {p: 32,  d: 0, i: 0.1, i_clamp: 1}
      joint2_t: {p: 32,  d: 0, i: 0.1, i_clamp: 1}

    state_publish_rate:  20          # Override default
    action_monitor_rate: 20            # Override default
    stop_trajectory_duration: 0        # Override default

  claw_trajectory_controller:
    type: effort_controllers/JointTrajectoryController
    publish_rate: 20
    joints:
      - joint_garra_ap
      - joint_garra_tracao_f1
      - joint_garra_tracao_f2
      - joint_garra_tracao_t1
      - joint_garra_tracao_t2

    gains: # Required because we're controlling an effort interface
      joint_garra_ap: {p: 32,  d: 0, i: 0.1, i_clamp: 1}
      joint_garra_tracao_f1: {p: 32,  d: 0, i: 0.1, i_clamp: 1}
      joint_garra_tracao_f2: {p: 32,  d: 0, i: 0.1, i_clamp: 1}
      joint_garra_tracao_t1: {p: 32,  d: 0, i: 0.1, i_clamp: 1}
      joint_garra_tracao_t2: {p: 32,  d: 0, i: 0.1, i_clamp: 1}

    state_publish_rate:  20          # Override default
    action_monitor_rate: 20            # Override default
    stop_trajectory_duration: 0        # Override default

  traction_trajectory_controller:
    type: velocity_controllers/JointTrajectoryController
    publish_rate: 20
    joints:
      - joint_eixo_tracao_ap
      - joint_eixo_tracao_f1
      - joint_eixo_tracao_f2
      - joint_eixo_tracao_t1
      - joint_eixo_tracao_t2

    gains: # Required because we're controlling an effort interface
      joint_eixo_tracao_ap: {p: 32,  d: 0, i: 0.1, i_clamp: 1}
      joint_eixo_tracao_f1: {p: 32,  d: 0, i: 0.1, i_clamp: 1}
      joint_eixo_tracao_f2: {p: 32,  d: 0, i: 0.1, i_clamp: 1}
      joint_eixo_tracao_t1: {p: 32,  d: 0, i: 0.1, i_clamp: 1}
      joint_eixo_tracao_t2: {p: 32,  d: 0, i: 0.1, i_clamp: 1}

    state_publish_rate:  20          # Override default
    action_monitor_rate: 20            # Override default
    stop_trajectory_duration: 0        # Override default

  # Publish all joint states -----------------------------------
  #joint_state_controller:
    #type: joint_state_controller/JointStateController
    #publish_rate: 20

  #traction_1c_controller:
    #type: velocity_controllers/JointVelocityController
    #joint: joint_eixo_tracao_ap
    #pid: {p: 32.0, i: 0.01, d: 0.0}

  #traction_1f_controller:
    #type: velocity_controllers/JointVelocityController
    #joint: joint_eixo_tracao_f1
    #pid: {p: 32.0, i: 0.01, d: 0.0}

  #traction_2f_controller:
    #type: velocity_controllers/JointVelocityController
    #joint: joint_eixo_tracao_f2
    #pid: {p: 32.0, i: 0.01, d: 0.0}

  #traction_1b_controller:
    #type: velocity_controllers/JointVelocityController
    #joint: joint_eixo_tracao_t1
    #pid: {p: 32.0, i: 0.01, d: 0.0}

  #traction_2b_controller:
    #type: velocity_controllers/JointVelocityController
    #joint: joint_eixo_tracao_t2
    #pid: {p: 32.0, i: 0.01, d: 0.0}
